core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -614,18 +613,3 @@ class NoOpMessageFormatter extends MessageFormatter {
   def writeTo(consumerRecord: ConsumerRecord[Array[Byte], Array[Byte]], output: PrintStream): Unit = {}
 }
 
-class ChecksumMessageFormatter extends MessageFormatter {
```

Comment:
I wonder if we should explicitly mention this one in the upgrade notes. It could be considered as part of the `console-consumer` tool which is part of our public API. It is in a gray zone. I am pretty sure that it is not used anymore but how knows... What's you take on this?

Reply:
Good question, I was debating it myself. Technically, this is not public API since it's not mentioned in the public javadoc and it's in the `core` module (everything is private in the `core` module).

However, it is possible that people would pass this class when using the console consumer. The checksum value has never been correct for message format 2, which has been the default since 0.11, so I figured the chances that people actually used this semi public API is low. Seems like there isn't much harm in mentioning it in the upgrade notes though, so will add.